internal/node/node.bzl

```diff
@@ -169,7 +169,7 @@ def _nodejs_binary_impl(ctx):
 
     script_path = _to_manifest_path(ctx, ctx.outputs.loader)
 
-    env_vars = "export BAZEL_TARGET=%s\n" % ctx.label
+    env_vars = ""
```

Comment:
hmm I am pretty sure we need this in a non-test context somewhere which is why we had to add it. something like running the golden test .accept bin

Reply:
Actually that works as with both `bazel test //golden/file:test` and `bazel run //golden/file:test` the `TEST_*` vars are set as it is a test target so the output
```
Update the golden file:

            bazel run ${debug ? '--compilation_mode=dbg ' : ''}${
          process.env['TEST_TARGET'].replace(/_bin$/, '')}.accept
```
works in both cases.

`bazel run //golden/file:test.accept` doesn't have the `TEST_*` set but it doesn't need them.

So in our repo we don't need `BAZEL_TARGET` set for any non-test targets.

I guess the question is if someone else is relying on this and if we should remove it or leave it and document it as a feature?

It can be useful in the run context to know which target is being run. A few others come to mind that would also be useful such as:
* `BAZEL_WORKSPACE` (to match `TEST_WORKSPACE`)
* `BAZEL_BINARY` (to match `TEST_BINARY`)
* `BAZEL_SRCDIR` (to match `TEST_SRCDIR` which is equal to `RUNFILES`)

Updating Angular to 1.0.0 yesterday I ran into the case where I needed all of these `TEST_*` vars to load the `require.resolve` loader patches in a bootstrap script since the bootstrap scripts are now being loaded with `--node_options=--require=...` and we're in an in-between state where the linker is not yet turned on for jasmine_node_test targets by default.

`/tools/testing/init_node_spec.ts` in https://github.com/angular/angular/pull/34589:
```
if (process.env['TEST_SRCDIR']) {
  // bootstrap the bazel require resolve patch since this
  // script is a bootstrap script loaded with --node_options=--require=...
  const path = require('path');
  require(path.posix.join(
      process.env['TEST_SRCDIR'], process.env['TEST_WORKSPACE'],
      (process.env['TEST_BINARY'] as string).replace(/\.(sh|bat)$/, '_loader.js'), ));
}

import 'zone.js/lib/node/rollup-main';
...
```

Reply:
Ohhhh. The linker relies on `TEST_*` targets right now which would be a problem for non-test targets:
```
    const wksp = env['TEST_WORKSPACE'];
    const target = env['TEST_TARGET'];
    if (!!wksp && !!target) {
      // //path/to:target -> //path/to
      const pkg = target.split(':')[0];
      this.packagePath = path.posix.join(wksp, pkg);
    }

    // If under runfiles (and not unit testing) then don't add a bin to target for bin links
    this.noBin = !!env['TEST_TARGET'] && wksp !== 'build_bazel_rules_nodejs' &&
        target !== '//internal/linker/test:unit_tests';
```

Seems like the right way to go here is to set at least `BAZEL_WORKSPACE`, `BAZEL_TARGET` and document them. Possibly `BAZEL_SRCDIR` and `BAZEL_BINARY` as well.